### PR TITLE
[AMS-274] JWT Bearer authentication added

### PR DESCRIPTION
**Pull request by denisivan0v**

### CODE REVIEW

File: src/VStore.Host/Startup.cs [hunk 1]
```diff
@@ -188,6 +213,33 @@ public void ConfigureServices(IServiceCollection services)
...
+                        TokenValidationParameters =
+                            new TokenValidationParameters
+                                {
+                                    ValidateIssuer = true,
+                                    ValidIssuer = jwtOptions.Issuer,
+
+                                    ValidateAudience = false,
+
+                                    ValidateIssuerSigningKey = true,
+                                    IssuerSigningKey = new SymmetricSecurityKey(Encoding.ASCII.GetBytes(jwtOptions.SecretKey)),
```
Comment (xakep139) [inline src/VStore.Host/Startup.cs]: Not UTF-8?

File: src/VStore.Host/Startup.cs [hunk 2]
```diff
@@ -188,6 +213,33 @@ public void ConfigureServices(IServiceCollection services)
...
+
+                                    ValidateLifetime = false,
```
Comment (xakep139) [inline src/VStore.Host/Startup.cs]: False?

File: src/VStore.Host/Startup.cs [hunk 3]
```diff
@@ -188,6 +213,33 @@ public void ConfigureServices(IServiceCollection services)
...
+                                    LifetimeValidator = (notBefore, expires, securityToken, validationParameters) =>
+                                                            {
+                                                                var utcNow = DateTime.UtcNow;
+                                                                return notBefore <= utcNow && utcNow <= expires;
```
Comment (xakep139) [inline src/VStore.Host/Startup.cs]: Should be
```csharp
return !(notBefore > utcNow || utcNow > expires);
```
due to `Nullable<DateTime>`

**Comment by xakep139** [comment]
Authentication on file uploading doesn't needed

File: src/VStore.Host/Startup.cs (hunk 1 above)
Comment (denisivan0v) [inline src/VStore.Host/Startup.cs]: Yes, we're going to support both upper and lower case letters, punctuation marks, and symbols (such as @, &, and *, for example) only.

File: src/VStore.Host/Startup.cs (hunk 2 above)
Comment (denisivan0v) [inline src/VStore.Host/Startup.cs]: Yes. Current OkAPI implementation doesn't include any lifetime information in JWT token

File: src/VStore.Host/Startup.cs (hunk 3 above)
Comment (denisivan0v) [inline src/VStore.Host/Startup.cs]: Good point.

**Comment by denisivan0v** [comment]
> Authentication on file uploading doesn't needed

Done

File: src/VStore.Host/Startup.cs (hunk 2 above)
Comment (xakep139) [inline src/VStore.Host/Startup.cs]: I saw `exp` field in their JWT payload data

File: src/VStore.Host/Startup.cs [hunk 4]
```diff
@@ -47,9 +49,19 @@
 
 namespace NuClear.VStore.Host
 {
-    // ReSharper disable once ClassNeverInstantiated.Global
+   // ReSharper disable once ClassNeverInstantiated.Global
```
Comment (xakep139) [inline src/VStore.Host/Startup.cs]: It's unacceptable

File: src/VStore.Host/Json/Int64ToStringJsonConverter.cs [hunk 5]
```diff
@@ -0,0 +1,23 @@
...
+
+using Newtonsoft.Json;
+
+namespace NuClear.VStore.Host.Json
+{
+    public sealed class Int64ToStringJsonConverter : JsonConverter
+    {
+        public override bool CanConvert(Type objectType) => objectType == typeof(long);
+
+        public override bool CanRead => false;
```
Comment (xakep139) [inline src/VStore.Host/Json/Int64ToStringJsonConverter.cs]: `CanWrite => true`?

File: src/VStore.Host/Startup.cs [hunk 6]
```diff
@@ -101,11 +113,10 @@ public void ConfigureServices(IServiceCollection services)
 
                                 settings.Culture = CultureInfo.InvariantCulture;
                                 settings.ContractResolver = new CamelCasePropertyNamesContractResolver();
-                                settings.Converters.Insert(0, new StringEnumConverter { CamelCaseText = true });
-                                settings.Converters.Insert(1, new ElementDescriptorJsonConverter());
-                                settings.Converters.Insert(2, new ElementDescriptorCollectionJsonConverter());
-                                settings.Converters.Insert(3, new TemplateDescriptorJsonConverter());
-                                settings.Converters.Insert(4, new ObjectDescriptorJsonConverter());
+                                for (var index = 0; index < CustomConverters.Length; index++)
```
Comment (xakep139) [inline src/VStore.Host/Startup.cs]: Can be `++index`

File: src/VStore.Host/Startup.cs (hunk 4 above)
Comment (denisivan0v) [inline src/VStore.Host/Startup.cs]: :see_no_evil:

File: src/VStore.Host/Json/Int64ToStringJsonConverter.cs (hunk 5 above)
Comment (denisivan0v) [inline src/VStore.Host/Json/Int64ToStringJsonConverter.cs]: This is by default

**Pull request merged**